backend/src/routes/api/namespaces/namespaceUtils.ts

```diff
@@ -60,13 +61,19 @@ export const applyNamespaceChange = async (
     throw createCustomError('Forbidden', "You don't have the access to update the namespace", 403);
   }
 
+  const disableServiceMesh = getDashboardConfig().spec.dashboardConfig.disableServiceMesh;
+
   let labels = {};
+  let annotations = {};
   switch (context) {
     case NamespaceApplicationCase.DSG_CREATION:
       labels = {
         'opendatahub.io/dashboard': 'true',
         'modelmesh-enabled': 'true',
       };
+      annotations = {
+        'opendatahub.io/service-mesh': String(!disableServiceMesh),
```

Comment:
As @andrewballantyne mentioned before, this is problematic because it will get both `false` and `undefined`.

We need to implement something like `featureFlagEnabled` in the backend: https://github.com/opendatahub-io/odh-dashboard/blob/3843c8222bc330feb44d0e0cf08ae37747ae7467/frontend/src/utilities/utils.ts#L8-L9

Reply:
I've added `featureFlagEnabled()` to the backend and used it in all places where we were previously pulling the value directly. Take a look here 93aa3ae